src/5e-SRD-Features.json

```diff
@@ -6953,7 +6953,7 @@
     ],
     "feature_specific": {
       "subfeature_options": {
-        "choose": 2,
+        "choose": "invocations known",
```

Comment:
Actually. Thinking about this more. I have a concern with this now being either an int or a string. Why would we want to do this?

Reply:
What does this get us?

Reply:
I don't really know what other value to use in this instance. The number of invocations available to the warlock vary by level. While I'd certainly prefer to use an int, I don't exactly see how I can have choose equal the number of invocations known as an int as any one value I use will be incorrect.

Reply:
I'm not sure if this change is a good idea in it's current state. I think this introduces inconsistency in the choice object that we were trying to get rid of with the Choice Refactor.

I wonder if a different way to approach this problem but have a similar outcome, is a way to extract out the list of Invocations into a list. That way you don't have to repeat it for every type? Then you can still have the choice for each level and the different number of invocations known. We're essentially doing the same thing as Known Spells, except it's a list of sub features instead of spells.

@fergcb?

Reply:
Keep in mind that some invocations have levels in their prerequisites.

Reply:
That is true. But prereqs are already set in each invocation. So that kind of handles itself?

Reply:
I may not have followed your suggestion then. Nevermind.

Regardless, I'll ping @fergcb after work to ask his opinion on the PR.

Reply:
I think using a string in the `choose` field isn't a great solution. That said, I can't think of a solution that would be both intuitive and good for deserialisation.

Until we think of such a solution, my vote would just be to leave it as a list of all the invocations, with their prerequisites, and leave it to clients to implement the functionality.

Reply:
> Until we think of such a solution, my vote would just be to leave it as a list of all the invocations, with their prerequisites,

By this do you mean ditching the choice structure entirely and just have it be a list of the invocations? Perhaps giving the property on `feature_specific` its own name, like `invocations`?

Reply:
> By this do you mean ditching the choice structure entirely and just have it be a list of the invocations? Perhaps giving the property on `feature_specific` its own name, like `invocations`?

Yeah, exactly that.

Reply:
I've done the suggested change.

Reply:
So on the one hand, we have gotten rid of the string. However, on the other hand, we've gotten rid of the choice completely as well as the number of invocations you can choose at each level, which is not ideal either as we have now lost data.

I have an idea. Invocations are, essentially, like picking spells. Would it make sense to add a `invocations_known` field in the `class_specific` field of Levels? I can't think of a better place for this to live. This would both accomplish what @abias1122 is trying to resolve by cleaning up Features, but also we keep all of the current data portrayed.

I think this is also a breaking change as well.

@fergcb, thoughts?

Reply:
Yeah, sorry I should have clarified more when I suggested we swap the Choice for just a list of invocations. I had indeed figured that we would also be adding an `invocations_known` attribute somewhere.

Putting it in the `class_specific` object makes the most sense, I think. The only other alternative, I suppose, would be to have it as a single map in the Eldritch Invocations feature itself, but I don't like that nearly as much.

Reply:
@bagelbits @fergcb `invocations_known` already exists on level class specific, so we should be good already.

<img width="1060" alt="invocations known" src="https://user-images.githubusercontent.com/28871516/190129970-b161e955-62d5-403f-b243-1d1cd00bfaaf.png">

Reply:
Here's a bit of the data. Notice that there are 20 documents with `invocations_known`: one document for each of the 20 warlock levels. This is all in main btw.

<img width="877" alt="invocations known search" src="https://user-images.githubusercontent.com/28871516/190149623-4528ff75-fa30-4006-b2e1-f22b519e148d.png">

Reply:
Oh sweet! Thanks for pointing that out!